### PR TITLE
Add GSI monitor scripts

### LINKED ISSUE #967: Absorb GSI monitor scripts
**Description**
As part of the overarching goal of moving global workflow scripts out of component repositories and under global workflow, all of the scripts currently linked in GSI monitor should be moved to global workflow.

**Requirements**
All jobs, scripts, and ush scripts that are currently linked by global workflow added to the global workflow repo instead and removed from the ignore list.

**Acceptance Criteria (Definition of Done)**
No jobs, scripts, or ush scripts linked from GSI monitor.

**Dependencies**
None

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
**Description**
As part of the overarching goal of moving global workflow scripts out of component repositories and under global workflow, all of the scripts currently linked in GSI monitor are added to global workflow. Scripts have been updated to reflect standardization already made to other workflow scripts (preamble, no backticks, etc.). See [7aa637](https://github.com/NOAA-EMC/global-workflow/pull/969/commits/7aa637ca4c7856abd71889ac94d70ec84a504dcf) for those modifications specifically.

Related GSI-Montitor issue: NOAA-EMC/GSI-Monitor/pull/23

Fixes #967

**Type of change**

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
- [x] Cycled test on Hera
  
**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes

### CODE REVIEW

**Comment by WalterKolczynski-NOAA** [comment]
Still waiting for another cycle or two to complete on Hera.

File: .gitignore [hunk 1]
```diff
@@ -178,19 +170,9 @@ ush/global_chgres.sh
 ush/global_chgres_driver.sh
 ush/global_cycle.sh
 ush/global_cycle_driver.sh
-ush/minmon_xtrct_costs.pl
-ush/minmon_xtrct_gnorms.pl
-ush/minmon_xtrct_reduct.pl
 ush/mkwfsgbl.sh
-ush/ozn_xtrct.sh
 ush/radmon_ck_stdout.sh
```
Comment (KateFriedman-NOAA) [inline .gitignore]: Still ignoring `ush/radmon_ck_stdout.sh`? I also don't see this script coming over.

Comment (WalterKolczynski-NOAA) [reply .gitignore]: It wasn't in link_workflow.sh, which theoretically means we don't use it, so I didn't bring it over. Probably means it also doesn't need to be ignored though.

Comment (KateFriedman-NOAA) [reply .gitignore]: Okie dokie, yeah please also remove it from being ignored then. Thanks!

**Comment by WalterKolczynski-NOAA** [comment]
Added a companion issue to GSI-Monitor that removes the scripts added here.

**Comment by WalterKolczynski-NOAA** [comment]
So, after running this I realize we don't test these jobs in any of our current development layouts. @EdwardSafford-NOAA @aerorahul can this be tested (or at least mostly tested) otherwise, or just fix it later once we have it set up to run?

**Comment by EdwardSafford-NOAA** [comment]
@WalterKolczynski-NOAA I can run the test driver scripts.  It's not a robust test but it will be something.  I'll try to make that happen this morning and will advise.

**Comment by aerorahul** [comment]
From `config.vrfy`:

Screen Shot 2022-08-15 at 9 44 37 AM

Do these not trigger the GSI monitoring jobs/scripts?

**Comment by EdwardSafford-NOAA** [comment]
Unfortunately the test-driver scripts will require some work to run with these changes.  It won't be difficult but it's more than I can do quickly today.  

I don't know a great deal (yet) about the global-workflow, but from what I see @aerorahul is right -- the settings in config.vrfy should set the 3 monitor jobs to run.

**Comment by WalterKolczynski-NOAA** [comment]
> From `config.vrfy`:
> 
> Screen Shot 2022-08-15 at 9 44 37 AM
> 
> Do these not trigger the GSI monitoring jobs/scripts?

Oh, they might. gempak and awips get turned on in config.base, so I didn't think to look anywhere else. Will check.

**Comment by WalterKolczynski-NOAA** [comment]
> > From `config.vrfy`:
> > Screen Shot 2022-08-15 at 9 44 37 AM
> > Do these not trigger the GSI monitoring jobs/scripts?
> 
> Oh, they might. gempak and awips get turned on in config.base, so I didn't think to look anywhere else. Will check.

They don't appear to. I actually already had them on for my run and I don't see any monitoring jobs. Unless they are run as part of another job name (they shouldn't, since they have j-jobs).

**Comment by RussTreadon-NOAA** [comment]
`config.vrfy` defines `TANKverf` variables under the `VRFYRAD`, `VRFYMINMON`, and `VRFYOZN` sections.   I believe monitoring output is written to the `TANKverf` directories.  The rocoto workflow executes the monitoring jobs from `vrfy.sh`.    This differs from operations.  A check of `gdasvrfy.log` will show us if (a) the monitoring jobs were run and (b) where the output is saved.   

We (developers) run some jobs which operations does not run.   For other jobs which both ops and dev run, we sometimes run the jobs from difference places in the workflow and write results to different directory structures.

**Comment by aerorahul** [comment]
> `config.vrfy` defines `TANKverf` variables under the `VRFYRAD`, `VRFYMINMON`, and `VRFYOZN` sections. I believe monitoring output is written to the `TANKverf` directories. The rocoto workflow executes the monitoring jobs from `vrfy.sh`. This differs from operations. A check of `gdasvrfy.log` will show us if (a) the monitoring jobs were run and (b) where the output is saved.
> 
> We (developers) run some jobs which operations does not run. For other jobs which both ops and dev run, we sometimes run the jobs from difference places in the workflow and write results to different directory structures.

Thanks @RussTreadon-NOAA. Your feedback and in-depth knowledge of the system is greatly appreciated.
This part of the workflow is not exercised by all developers and so it remains somewhat unknown to most who do not exercise data assimilation components.  
We will hope to do this better, starting with perhaps not including a number of activities in the same `vrfy` job.  A lot gets swept under this one job.

@WalterKolczynski-NOAA 
When we run the next "cycled" test, see if we can run `vrfy` and capture some of the DA verification and monitoring jobs output.  We might need to run for a few cycles.
If you are running a test with this branch, I can take a look.  I think I remember enough to tell the difference if this is being exercised or not.

**Comment by WalterKolczynski-NOAA** [comment]
Okay, at least some of these scripts *are* running. Currently `exgdas_atmos_verfrad.sh` is dying on a failed move for abi_g16. I can add checks that the gunzip file exists before trying to rename and expand it, but are there any SATYPEs that should cause a fail if missing?

**Comment by EdwardSafford-NOAA** [comment]
Yikes!  Missing diag files are common and shouldn't cause the script to die, but rather be reported as missing in the warning report.  Can you point me to your log file?  I'd like to take a look.

**Comment by bubba0077** [comment]
> Yikes! Missing diag files are common and shouldn't cause the script to die, but rather be reported as missing in the warning report. Can you point me to your log file? I'd like to take a look.

This is just a side-effect of having `set -e` on now. I just wanted to confirm none of these should cause a failure.

**Comment by WalterKolczynski-NOAA** [comment]
> Yikes! Missing diag files are common and shouldn't cause the script to die, but rather be reported as missing in the warning report. Can you point me to your log file? I'd like to take a look.

This is just a side-effect of having set -e on now. I just wanted to confirm none of these should cause a failure.

**Review by EdwardSafford-NOAA — APPROVED** [review]
Looks ok to me.

**Comment by WalterKolczynski-NOAA** [comment]
Finally worked through all the issues due to undefined variables or operating on missing files.

@EdwardSafford-NOAA Please carefully review specifically those changes in the most recent commit (cb29351) to make sure I haven't caused any unwanted behavior with my changes.

File: ush/radmon_verf_angle.sh [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+EOF
+
+	 startmsg
+         ./${angle_exec} < input >>   ${pgmout} 2>>errfile
+         export err=$?; err_chk
+         if [[ $err -ne 0 ]]; then
+             fail=$(expr $fail + 1)
+         fi
+
+#-------------------------------------------------------------------
```
Comment (EdwardSafford-NOAA) [inline ush/radmon_verf_angle.sh]: This is on me -- the comments from 195-209 are out of date and can be removed.

File: scripts/exgdas_atmos_verfrad.sh [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+#  Other variables
+export RAD_AREA=${RAD_AREA:-glb}
+export MAKE_CTL=${MAKE_CTL:-1}
+export MAKE_DATA=${MAKE_DATA:-1}
+export USE_ANL=${USE_ANL:-1}
+export PDATE=${PDY}${cyc}
+export DO_DIAG_RPT=${DO_DIAG_RPT:-1}
+export DO_DATA_RPT=${DO_DATA_RPT:-1}
+export USE_MAIL=${USE_MAIL:-0}
```
Comment (EdwardSafford-NOAA) [inline scripts/exgdas_atmos_verfrad.sh]: The capability to mail warning reports directly from the data extraction was removed a while ago so the USE_MAIL, MAIL_TO, and MAIL_CC variables can be removed.

File: ush/radmon_verf_time.sh [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+RAD_AREA=${RAD_AREA:-glb}
+REGIONAL_RR=${REGIONAL_RR:-0}
+rgnHH=${rgnHH:-}
+rgnTM=${rgnTM:-}
+SATYPE=${SATYPE:-}
+MAIL_TO=${MAIL_TO:-}
+MAIL_CC=${MAIL_CC:-}
+VERBOSE=${VERBOSE:-NO}
+LITTLE_ENDIAN=${LITTLE_ENDIAN:-0}
+USE_MAIL=${USE_MAIL:-1}
```
Comment (EdwardSafford-NOAA) [inline ush/radmon_verf_time.sh]: USE_MAIL, MAIL_TO, and MAIL_CC can be removed.

**Review by EdwardSafford-NOAA — APPROVED** [review]
This looks good and thank you for making many improvements.  I noted some minor things that can be removed, either now or as a later step.

File: ush/radmon_verf_angle.sh (hunk 2 above)
Comment (WalterKolczynski-NOAA) [reply ush/radmon_verf_angle.sh]: Removed

File: scripts/exgdas_atmos_verfrad.sh (hunk 3 above)
Comment (WalterKolczynski-NOAA) [reply scripts/exgdas_atmos_verfrad.sh]: Removed

File: ush/radmon_verf_time.sh (hunk 4 above)
Comment (WalterKolczynski-NOAA) [reply ush/radmon_verf_time.sh]: Removed

**Review by KateFriedman-NOAA — APPROVED** [review]

**Pull request merged**